tests/manage/test_pvc_delete_verify_size_is_returned_to_backendpool.py

```diff
@@ -0,0 +1,257 @@
+    """
+    Create a storage class
+    """
+    template_data['rbd_storageclass_name'] = sc_name
+    template_data['rbd_pool'] = pool_name
+    create_oc_resource(
+        'storageclass.yaml', TEMPLATES_DIR, _templating,
+        template_data, "CSI/rbd"
+    )
+
+    # Validate storage class created
+    assert SC.get(resource_name=sc_name)
+
+
+def create_secret(secret_name, pool_name):
+    """
+    Create secret to store username and password
+    """
+
+    # Get the keyring for admin
+    admin = ocp.getbase64_ceph_secret("client.admin")
+    template_data['base64_encoded_admin_password'] = admin
+
+    # Create a user=kubernetes
+    assert ocp.exec_ceph_cmd(
+        ceph_cmd=f'ceph auth get-or-create-key client.kubernetes mon '
+        f' "allow profile rbd" osd "profile rbd pool={pool_name}"')
+
+    # Get the keyring for user=kubernetes
+    kubernetes = ocp.getbase64_ceph_secret("client.kubernetes")
+    template_data['base64_encoded_user_password'] = kubernetes
+
+    template_data['csi_rbd_secret'] = secret_name
+
+    create_oc_resource(
+        'secret.yaml', TEMPLATES_DIR, _templating, template_data, "CSI/rbd"
+    )
+
+    # Validate secret is created
+    assert SECRET.get(resource_name=secret_name)
+
+
+def check_ceph_used_space():
+    """
+    Check for the used space in cluster
+    """
+
+    cmd = "ceph status"
+    ceph_status = ocp.exec_ceph_cmd(cmd)
+    assert ceph_status is not None
+    used = ceph_status['pgmap']['bytes_used']
+    GB = (1024 ** 3)
+    used_in_gb = used / GB
+    return used_in_gb
+
+
+def create_pvc(pvc_name, sc_name, pool_name):
+    """
+    Create a pvc
+    """
+    template_data['pvc_name'] = pvc_name
+    template_data['user_namespace'] = PROJECT_NAME
+    template_data['rbd_storageclass_name'] = sc_name
+    template_data['pvc_size'] = '10Gi'
+    create_oc_resource(
+        'pvc.yaml', TEMPLATES_DIR, _templating, template_data, "CSI/rbd"
+    )
+
+    # Validate pvc is in bound state
+    assert PVC.wait_for_resource(condition="Bound", resource_name=pvc_name)
+
+    # Validate pvc is created on ceph
+    pvc_info = PVC.get(resource_name=pvc_name)
+
+    name = get_ceph_tools_pod()
+    po = pod.Pod(name, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+    cmd = f"rbd ls -p {pool_name}"
+    pvc, _, _ = po.exec_command(cmd=cmd, timeout=20)
+
+    assert pvc_info['spec']['volumeName'] in pvc
+
+
+def create_pod(pod_name, pvc_name):
+    """
+    Create a app pod
+    """
+
+    template_data['pod_name'] = pod_name
+    template_data['pvc_name'] = pvc_name
+    create_oc_resource(
+        'pod.yaml', TEMPLATES_DIR, _templating, template_data, "CSI/rbd"
+    )
+
+    assert POD.wait_for_resource(condition="Running", resource_name=pod_name)
+
+
+def run_io(pod_name):
+    """
+    Run io on the mount point
+    """
+
+    run_cmd(
+        f"oc rsh -n {PROJECT_NAME} {pod_name}"
+        " dd if=/dev/urandom of=/var/lib/www/html/dd_ar bs=10M count=950"
+    )
+
```

Comment:
@Akarsha-rai are we confirming that the file is indeed written in the mount point ?

Reply:
done